atlassian_jwt_auth/contrib/django/middleware.py

```diff
@@ -1,21 +1,17 @@
 from django.conf import settings
 
-import atlassian_jwt_auth
-from ..server.helpers import _requires_asap
-from .utils import parse_jwt, verify_issuers, _build_response
-from .decorators import _get_verifier
+from atlassian_jwt_auth.frameworks.django.middleware import (
+    OldStyleASAPMiddleware
+)
 
 
-class ASAPForwardedMiddleware(object):
+class ProxiedAsapMiddleware(OldStyleASAPMiddleware):
     """Enable client auth for ASAP-enabled services that are forwarding
     non-ASAP client requests.
 
-    This must come before any authentication middleware.
-
-    DEPRECATED: use ASAPMiddleware instead.
```

Comment:
So this is no longer deprecated?

Reply:
I think there's two use cases to talk about here:
- Django middleware which performs ASAP auth on all requests 
- Django middleware with weird client proxy behavior

Previously, the two were combined into the same middleware (which had two versions) which made the first and more common case unnecessarily complicated. I split out the simple use-case into an old-style and new-style middleware in `frameworks/django`. I then refactored and combined the two middleware in `contrib/django/middleware.py` into a single `ProxiedAsapMiddleware` that inherits from the simple middleware in `frameworks/django`.

The main idea here is... I flattened the odd contrib custom middleware (as long as we're cleaning up this tech debt anyways) and created a new simple middleware based off the shared common decorator logic in frameworks/django.

So, to answer your original question, the deprecated middleware doesn't really exist anymore but the diff makes it look more confusing than it is.